src/Data/Repositories/APITokenRepository.cs

```diff
@@ -21,6 +21,12 @@ public class APITokenRepository : IAPITokenRepository
         _dbContextFactory = dbContextFactory;
     }
     
+    private async Task<T> WithContext<T>(Func<ApplicationDbContext, Task<T>> func)
+    {
+        await using var context = _dbContextFactory.CreateDbContext();
+        return await func(context);
+    }
+    
```

Comment:
@Jossec101 check this. I think it is a fancier method for accessing to the db context instead of calling all the time to this function. You can see usage in `GetByToken` function. I just wanted to try this way and show to the team something different but if it's too much or you don't like it i can just go back to the usual :)

Reply:
I don't really see the advantage of using callbacks/delegates with generics here, if you ask me what's easier a repeated line to get the context async or a generic method, I will rather go for the 1st one, besides;
```suggestion
    private async Task<T> WithContext<T>(Func<ApplicationDbContext, Task<T>> func)
    {
        await using var context = await _dbContextFactory.CreateDbContextAsync();
        return await func(context);
    }
    
```

Also the code below makes debugging harder if the return is not directly a variable (breakpoints are useless in this case)
:

```return await WithContext(
            async context =>
            {
                var result = await context.ApiTokens.FirstOrDefaultAsync(x => x.TokenHash == token);
                return result;
            }

            );
    }
```

I think this is not KISS, `keep it stupid simple`